Riot/Modules/Room/Views/BubbleCells/Styles/Bubble/BubbleRoomTimelineCellDecorator.swift

```diff
@@ -183,21 +205,41 @@ class BubbleRoomTimelineCellDecorator: PlainRoomTimelineCellDecorator {
     
     // MARK: - Private
     
-    private func createTimestampLabel(cellData: MXKRoomBubbleCellData, bubbleComponent: MXKRoomBubbleComponent, viewTag: Int) -> UILabel {
+    // MARK: Timestamp management
+    
+    private func createTimestampLabel(cellData: MXKRoomBubbleCellData, bubbleComponent: MXKRoomBubbleComponent, viewTag: Int, textColor: UIColor) -> UILabel {
         
         let timeLabel = UILabel()
 
         timeLabel.text = cellData.eventFormatter.timeString(from: bubbleComponent.date)
         timeLabel.textAlignment = .right
-        timeLabel.textColor = ThemeService.shared().theme.textSecondaryColor
-        timeLabel.font = UIFont.systemFont(ofSize: 11, weight: .light)
+        timeLabel.textColor = textColor
+        timeLabel.font = self.theme.fonts.caption2
         timeLabel.adjustsFontSizeToFitWidth = true
         timeLabel.tag = viewTag
         timeLabel.accessibilityIdentifier = "timestampLabel"
         
         return timeLabel
     }
     
+    func createTimestampLabel(for cellData: RoomBubbleCellData) -> UILabel? {
+        return self.createTimestampLabel(for: cellData, textColor: self.theme.textSecondaryColor)
+    }
+    
+    private func createTimestampLabel(for cellData: RoomBubbleCellData, textColor: UIColor) -> UILabel? {
+        
+        let componentIndex = cellData.mostRecentComponentIndex
+        
+        guard let bubbleComponents = cellData.bubbleComponents,
+              componentIndex < bubbleComponents.count else {
+                  return nil
+              }
```

Comment:
The indentation of the return and closing brace seems off to me, but I would normally put `else {` on a new line, so not sure if this is what Xcode does anyway.

Reply:
Yes it depends on people but we can use the same rules. On my side I prefer line return for `else` of the `guard`, to see quickly the return without reading all the conditions of the guard.